### ISSUE TITLE
uninitialized model throws: Attempt to execute code removed by Dart AOT compiler (TFA) when compiled to exe

### ISSUE THREAD

**Issue by KhaledBasiony** [realm/realm-dart#1415]
### What happened?
I need to have an uninitialized model (let's call it `Product`) like this:

```dart
final Product prod;

if (something) {
    prod = Product('prod 1');
} else {
    prod = Product('prod 2');
}
```

the problem appears when I compile the app using `dart compile exe ...` and run it, it throws that error:
`Attempt to execute code removed by Dart AOT compiler (TFA)`

Notes:

* It DOESN'T work whenever I specify the type `Product`
* It DOES work if initialize it immediately (`final prod = Product('prod');`)
* It DOES work if I don't use a type (`final prod;` for example)

I think I can work around it for now, I am just hoping this would get fixed :grinning:

if this not the appropriate place for this issue please guide me 

### Repro steps
1. `dart create dummy_project`
1. `dart pub add realm_dart`
1. `dart run realm_dart install`
1. create a file `lib/model.dart` with the following contents:
```dart
import 'package:realm_dart/realm.dart';

    part 'model.g.dart';
    
    @RealmModel()
    class _Product {}
    ```1. `dart run realm_dart generate`
1. put the following code in `bin/dummy_project.dart`:
```dart
import 'package:dummy_project/model.dart';

    void main(List<String> arguments) {
      final Product prod;
      prod = Product();
      print(prod);
    }
    ```1. `dart compile exe bin/dummy_project.dart -o bin/out`
1. `./bin/out`
1. panic :/
```sh
Unhandled exception:
Attempt to execute code removed by Dart AOT compiler (TFA)
#0      main (file:///home/khaled/Documents/dummy_project/bin/dummy_project.dart)
#1      _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:294)
#2      _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:189)
```

### Version
Dart SDK version: 3.1.0 (stable) (Tue Aug 15 21:33:36 2023 +0000) on "linux_x64"

### What Atlas Services are you using?
Local Database only

### What type of application is this?
Dart standalone application

### Client OS and version
OS: openSUSE Tumbleweed x86_64

### Code snippets
```dart
final Product prod;

if (something) {
    prod = Product('prod 1');
} else {
    prod = Product('prod 2');
}
```

### Stacktrace of the exception/crash you're getting
```shell
Unhandled exception:
Attempt to execute code removed by Dart AOT compiler (TFA)
#0      main (file:///home/khaled/Documents/dummy_project/bin/dummy_project.dart)
#1      _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:294)
#2      _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:189)
```

**Comment by milindgoel15** [comment]
Related to https://github.com/dart-lang/sdk/issues/52874

Dont make the product final. Make it nullable 

`final Product prod;` to `Product? prod;`

**Comment by KhaledBasiony** [comment]
Hello @milindgoel15, thanks for the comment.

I already worked around it by using "late"

But really, I don't want to make it nullable when it's not, and I don't want to use "late" and lose the static analysis/lints, and obviously I don't want it to be dynamic.

I just thought it would be nice for this particular case to be solved.

Thanks anyway.

**Comment by milindgoel15** [comment]
You could do ternary check instead of assigning them in the if else.

`final Product product = something ? First : second`

**Comment by KhaledBasiony** [comment]
Huh, clever!

Unfortunately my actual code is a bit more complicated than a simple boolean check.

Will try to make it work tho, thanks!

**Comment by dotjon0** [comment]
> Huh, clever!
> 
> Unfortunately my actual code is a bit more complicated than a simple boolean check.
> 
> Will try to make it work tho, thanks!

If its a state, perhaps try https://riverpod.dev/, maybe a stream as then you can avoid nullable via an empty stream

**Comment by nielsenko** [comment]
There was a number of dart compiler bugs introduced with Dart 3 in relation to classes marked `Finalizable` (such as `RealmObject` subclasses). This is fixed by upgrade to 3.3 or later https://github.com/realm/realm-dart/pull/1648.
